File: ruby/space-age/space_age.rb
class SpaceAge
  attr_accessor :age
  def initialize(age)
    @age = age
  end

  def seconds
    @age 
  end

  def on_earth
    @age/31557600.round(2)
  end

  def on_mercury
    (on_earth/(0.2408467)).round(2)
  end

  def on_venus
    (on_earth/(0.61519726)).round(2)
  end

  def on_mars
    (on_earth/(1.8808158)).round(2)
  end

  def on_jupiter
    (on_earth/(11.862615)).round(2)
  end

  def on_saturn
    (on_earth/(29.447498)).round(2)
  end

  def on_uranus
    (on_earth/(84.016846)).round(2)
  end

  def on_neptune
    (on_earth/(164.79132)).round(2)
  end
end

File: javascript/space-age/space-age.js
var SpaceAge = function(seconds){
  this.seconds = seconds;
  this.earthTime = this.seconds/31557600.0;
}

SpaceAge.prototype.onEarth = function() {
  return round(this.earthTime, 100.0);
}

SpaceAge.prototype.onMercury = function() {
  return round(this.earthTime/0.2408467, 100.0);
}

SpaceAge.prototype.onVenus = function() {
  return round(this.earthTime/0.61519726, 100.0);
}

SpaceAge.prototype.onMars = function() {
  return round(this.earthTime/1.8808158, 100.0);
}

SpaceAge.prototype.onJupiter = function() {
  return round(this.earthTime/11.862615, 100.0);
}

SpaceAge.prototype.onSaturn = function() {
  return round(this.earthTime/29.447498, 100.0);
}

SpaceAge.prototype.onUranus = function() {
  return round(this.earthTime/84.016846, 100.0);
}

SpaceAge.prototype.onNeptune = function() {
  return round(this.earthTime/164.79132, 100.0);
}

function round(num, toDecimal){
  return Math.round(num * toDecimal)/toDecimal;
}

module.exports = SpaceAge;